src/Sarif/Writers/SarifLogger.cs

```diff
@@ -555,6 +555,7 @@ public void LogToolNotification(Notification notification)
 
             _run.Invocations[0].ToolExecutionNotifications = _run.Invocations[0].ToolExecutionNotifications ?? new List<Notification>();
             _run.Invocations[0].ToolExecutionNotifications.Add(notification);
+            _run.Invocations[0].ExecutionSuccessful = true;
```

Comment:
>ExecutionSuccessful [](start = 32, length = 19)

You need to check the notification level here. If it is error then ExecutionSuccessful is set to false.